docs/json_schemas/dns_record/v0/requirer.json

```diff
@@ -0,0 +1,261 @@
+{
+  "$defs": {
+    "BaseModel": {
+      "properties": {},
+      "title": "BaseModel",
+      "type": "object"
+    },
+    "DnsRecordRequirer": {
+      "description": "List of domains for the provider to manage.",
+      "properties": {
+        "dns_domains": {
+          "anyOf": [
+            {
+              "items": {
+                "$ref": "#/$defs/RequirerDomains"
+              },
+              "type": "array"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "List of domains for the provider to manage.",
+          "title": "Dns Domains"
+        },
+        "dns_entries": {
+          "anyOf": [
+            {
+              "items": {
+                "$ref": "#/$defs/RequirerEntries"
+              },
+              "type": "array"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "List of DNS records for the provider to manage.",
+          "title": "Dns Entries"
+        }
+      },
+      "required": [
+        "dns_domains",
+        "dns_entries"
+      ],
+      "title": "DnsRecordRequirer",
+      "type": "object"
+    },
+    "RecordClass": {
+      "description": "Represent the DNS record classes.",
+      "enum": [
+        "IN",
+        "CH",
+        "HS"
+      ],
+      "title": "RecordClass",
+      "type": "string"
+    },
+    "RecordType": {
+      "description": "Represent the DNS record types.",
+      "enum": [
+        "A",
+        "AAAA",
+        "CNAME",
+        "MX",
+        "DKIM",
+        "SPF",
+        "DMARC",
+        "TXT",
+        "CAA",
+        "SRV",
+        "SVCB",
+        "HTTPS",
+        "PTR",
+        "SOA",
+        "NS",
```

Comment:
I'm wondering if these two should be supported, as they're very close to the domain definition

Reply:
My approach was to support everything as defined in the standard, but we can drop support if you think that's more reasonble